prolif/io/protein_helper.py

```diff
@@ -0,0 +1,594 @@
+import logging
+import warnings
+from pathlib import Path
+
+from rdkit import Chem
+from rdkit.Chem import AllChem
+
+from prolif.constants import (
+    AMBER_POOL,
+    ATOMNAME_ALIASES,
+    CHARMM_POOL,
+    FORMAL_CHARGE_ALIASES,
+    GROMOS_POOL,
+    MAX_AMIDE_LENGTH,
+    OPLS_AA_POOL,
+    RESNAME_ALIASES,
+    STANDARD_AA,
+    STANDARD_RESNAME_MAP,
+    TERMINAL_OXYGEN_NAMES,
+)
+from prolif.molecule import Molecule
+from prolif.residue import Residue, ResidueGroup
+
+logger = logging.getLogger(__name__)
+
+
+class ProteinHelper:
+    """ProteinHelper is a class to standardize the residue names and fix the bond order
+    when reading the non-standard residues with RDKit for a molecule.
+
+    Parameters
+    ----------
+    input_topology : str or Molecule
+        The path of input protein molecule (.pdb) or a ProLIF Molecule.
+
+    Attributes
+    ----------
+    protein_mol : Molecule
+        The ProLIF Molecule instance representing the protein.
+
+    Notes
+    -----
+    This class is designed to work with ProLIF Molecule instances or PDB files.
+    It reads the input topology, standardizes the residue names, and fixes the bond
+    orders based on the provided templates or standard amino acid template.
+    However, the explicit hydrogen atoms will be removed from the molecule.
+
+    Example
+    -------
+    >>> from prolif.io import ProteinHelper
+    >>> protein_helper = ProteinHelper("path/to/protein.pdb")
+    >>> protein_helper.standardize_protein()
+    >>> # Now, protein_helper.protein_mol contains the standardized protein molecule.
+    >>> # You can access the residues using protein_helper.protein_mol.residues.
+
+    """
+
+    def __init__(self, input_topology: Molecule | str | Path):
```

Comment:
Thinking about this a bit more, I think we should invert the input params between the initialization and the standardize method. The protein-helper should be configured with the templates, and then ran on a specific input, rather than the other way around